apps/rush-lib/src/api/PackageJsonEditor.ts

```diff
@@ -217,6 +217,7 @@ export class PackageJsonEditor {
 
   public saveIfModified(): boolean {
     if (this._modified) {
+      JsonFile.save(this._sanitize(), this._filePath, { updateExistingFile: true });
       JsonFile.save(this._normalize(), this._filePath, { updateExistingFile: true });
```

Comment:
We shouldn't be saving this file twice.

Reply:
Actually, saving twice is an easy fix.

The sort logic happens inside https://github.com/microsoft/rushstack/blob/master/apps/rush-lib/src/api/PackageJsonEditor.ts#L143. So the data passed to JsonFile.save is already sorted. The point of unsorted values comes from the jju package's update method, this method keeps the original order as much as possible.

The alternatives can be switch to another json updater module, or self implement this logic. Compare to those solutions, clearing current package.json's dependencies (save once), and jju.update sorted dependency mandatarily keeps the order (save twice) is such a quick fix.

If this fix is not in a right direction, i would like to stick to another solution. WDYT?